### DISCUSSION TITLE
theme-park theming

### DISCUSSION THREAD

**Discussion by deathbybandaid** [authelia/authelia#4517] [Ideas]
### Description

If not familiar with  [theme-park](https://github.com/GilbN/theme.park), it's a project tat aims to allow users to easily theme all of their selfhosted web applications.

[This authelia feature request](https://github.com/authelia/authelia/issues/1935) would easily be covered by incorporating the methodology of theme-park.

### Use Case

Authelia can have community created and updated themes.

### Details

theme-park uses "css base" files to convert the css of an application and create a uniformity that can be easily skinned.

### Documentation

Authelia is fairly simple in terms of the login page, as well as the authenticated page. There are some css elements that I missed, such as the little red bar in the upper right upon invalid login, and the username/password text. But this is mostly a demonstration. Most of the demonstration is the images below, as it's very easy and fast to change the themes.

I have the below working nicely with the plex theme from theme-park.

````
body {
    background: var(--main-bg-color);
    background-repeat: repeat, no-repeat;
    background-attachment: fixed, fixed;
    background-position: center center, center center;
    background-size: auto, cover;
    -webkit-background-size: auto, cover;
    -moz-background-size: auto, cover;
    -o-background-size: auto, cover;
    color: var(--text);
}

.MuiButton-root {
    background-color: var(--button-color);
    color: var(--button-color-text);
    font-weight: bold;
}
````

Stock:
![image](https://user-images.githubusercontent.com/18010545/206050178-f6ed554e-5722-4691-afb7-428f97001c0d.png)

plex:
![image](https://user-images.githubusercontent.com/18010545/206050285-b6ec2be5-b597-4972-b87f-5fe6a0eafa89.png)

nord:
![image](https://user-images.githubusercontent.com/18010545/206050338-3c6371c9-c47e-4c1b-baf2-058ab9774aa0.png)

aquamarine:
![image](https://user-images.githubusercontent.com/18010545/206050466-cf727be5-ec1d-4e0c-b23a-933f79c570ba.png)

hotline:
![image](https://user-images.githubusercontent.com/18010545/206050536-24a4ff5e-d71a-44b6-bfc7-ad0d2960a2af.png)

hotpink:
![image](https://user-images.githubusercontent.com/18010545/206050702-b9936f1c-b3d5-4215-a6d8-e2c535b725b7.png)

dracula:
![image](https://user-images.githubusercontent.com/18010545/206050794-a86f7ee5-ddb1-4347-90fa-02493038ec15.png)

dark:
![image](https://user-images.githubusercontent.com/18010545/206050858-06b9a4ce-dc66-4112-aeb1-4ea4ab848122.png)

organizr:
![image](https://user-images.githubusercontent.com/18010545/206050920-baf45479-1c16-4d9b-bc73-05eac9f8a38e.png)

space-gray:
![image](https://user-images.githubusercontent.com/18010545/206050995-772d0e35-4046-472f-9dc5-6992f1d6f61a.png)

overseerr:
![image](https://user-images.githubusercontent.com/18010545/206051089-c4143592-e1f8-4b95-b4ae-4456b9df2e4e.png)

maroon:
![image](https://user-images.githubusercontent.com/18010545/206051126-915e0dd5-9308-4508-a3ea-ed33b7f92eae.png)

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)
- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
As far as I can see this is already working. If this is not the case then please explain what we need to change to support this?

**Comment by deathbybandaid** [comment]
Nothing needs to change with authelias code. It just needs a little tweaking from somebody more familiar with authelias css, and some testing.

It doesn't matter to me who takes credit when PR-ing it to the theme-park repo, but it might be better coming straight from an authelia maintainer.

**Comment by james-d-elliott** [comment]
I've converted this to an idea discussion so others can get involved rather than it being an issue/fr

**Comment by jpdsc** [comment]
@deathbybandaid looks great! Are you planning on pushing a PR at TP Github for this? GilbN already confirm he will not be working on the themes for Authelia himself so if you push a PR using his guidelines I think he'll approve: https://docs.theme-park.dev/custom/

**Reply by deathbybandaid** [reply]
I think I'd need a deeper dive output of all available css options that I'd need to tweak to get it 100%, but yes, I could do a PR for this

**Comment by Ninja-FSE** [comment]
This looks amazing! Do you mind to write up a small howto on how you added theme-park to authelia?

**Reply by deathbybandaid** [reply]
Feel free!

**Comment by gravelfreeman** [comment]
Could you share the theme for the overseerr theme please? Thank you!

**Reply by deathbybandaid** [reply]
It works similar to all the other theme-park stuff. The CSS I provided here uses the standardized CSS variable names for theme-park. You simply have to import the theme file.
